### PR TITLE
Fix light card

### LINKED ISSUE #4251: Light card icon and slider
**Checklist:**

- [x] I updated to the latest version available
- [x] I cleared the cache of my browser

**Home Assistant release with the issue:**
0.102.0b3

**Last working Home Assistant release (if known):**
0.102.0b2
**UI (States or Lovelace UI?):**
LL

**Browser and Operating System:**
FF and Chrome

**Description of problem:**
Since the last beta the icon of the light card is a little to big and the knob of the slider is missing.

b2

b3

**Javascript errors shown in the web inspector (if applicable):**

```

```

**Additional information:**

### LINKED ISSUE #4256: Light card broken
**Checklist:**

- [x] I updated to the latest version available
- [x] I cleared the cache of my browser

**Home Assistant release with the issue:** 0.102.0

**Last working Home Assistant release (if known):** 0.102.0b2

**UI (States or Lovelace UI?):** Lovelace

**Browser and Operating System:** Android / Chrome, Linux / Firefox

**Description of problem:**
Light cards without a slider are not displayed properly anymore. New they are just boxes with text but without icon.
grafik

**Javascript errors shown in the web inspector (if applicable):**

```
No error
```

**Additional information:**

### PR DESCRIPTION
**Pull request by bramkragten**
Fixes https://github.com/home-assistant/home-assistant-polymer/issues/4251
Fixes https://github.com/home-assistant/home-assistant-polymer/issues/4256

### CODE REVIEW

File: src/panels/lovelace/cards/hui-light-card.ts [hunk 1]
```diff
@@ -100,30 +99,29 @@ export class HuiLightCard extends LitElement implements LovelaceCard {
             ${supportsFeature(stateObj, SUPPORT_BRIGHTNESS)
               ? html`
                   <round-slider
-                    min="1"
+                    min="0"
```
Comment (thomasloven) [inline src/panels/lovelace/cards/hui-light-card.ts]: This changes the function of the slider.
Now sliding to the bottom will turn the light off as opposed to setting it to minimum brightness.
This results in the handle disappearing, which might be confusing.

File: src/panels/lovelace/cards/hui-light-card.ts [hunk 2]
```diff
@@ -100,30 +99,29 @@ export class HuiLightCard extends LitElement implements LovelaceCard {
...
-            ></ha-icon>
+              })}
+              @click=${this._handleClick}
+            ></paper-icon-button>
           </div>
         </div>
 
         <div id="info">
-          <div class="brightness" @ha-click="${this._handleClick}">
+          <div class="brightness">
```
Comment (thomasloven) [inline src/panels/lovelace/cards/hui-light-card.ts]: Good!
I couldn't remember why this was here, so I left it in. But I really can't see any point to it at all.

File: src/panels/lovelace/cards/hui-light-card.ts [hunk 3]
```diff
@@ -266,24 +265,21 @@ export class HuiLightCard extends LitElement implements LovelaceCard {
...
+        color: var(--paper-item-icon-color, #44739e);
+        width: 60%;
+        height: auto;
+      }
+
       .slider-center {
         position: absolute;
-        width: 70%;
-        height: 70%;
-        max-height: calc(100% - 40px);
```
Comment (thomasloven) [inline src/panels/lovelace/cards/hui-light-card.ts]: Does this work with icons of different aspect ratios?

**Comment by thomasloven** [comment]
I'm not sure how it behaved before, but does the height of the card change between lights that support vs. don't support brightness?
I.e. will it look weird if you have two of them next to each other in a horizontal stack?

**Review by thomasloven — APPROVED** [review]

File: src/panels/lovelace/cards/hui-light-card.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-light-card.ts]: This fixes the handle disappearing, but yeah it does turn the light off when sliding all the way down. I think I prefer it this way. Do we have more options? Old behaviour was not turning off but still showing the handle?

**Comment by bramkragten** [comment]
> I'm not sure how it behaved before, but does the height of the card change between lights that support vs. don't support brightness?
> I.e. will it look weird if you have two of them next to each other in a horizontal stack?

Mmm yeah, it now looks like an entity button card when it doesn't support brightness...

File: src/panels/lovelace/cards/hui-light-card.ts (hunk 1 above)
Comment (thomasloven) [reply src/panels/lovelace/cards/hui-light-card.ts]: Min was 1 with the old slider too. https://github.com/home-assistant/home-assistant-polymer/pull/3634/files#diff-f2231026bfc2c43209404df9b7bbb679L35

It's probably better this way. I just didn't want it to slide by without comment.

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-light-card.ts]: But in the old version the handle did stay, is that something that was changed in `round-slider`?

File: src/panels/lovelace/cards/hui-light-card.ts (hunk 3 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-light-card.ts]: Icons are always square

File: src/panels/lovelace/cards/hui-light-card.ts (hunk 1 above)
Comment (thomasloven) [reply src/panels/lovelace/cards/hui-light-card.ts]: Probably. The slider is disabled if the value is outside of the range.
Maybe the old one drew it outside the bar, or just clamped it to the min-max range?

Comment (iantrich) [reply src/panels/lovelace/cards/hui-light-card.ts]: I'm not a fan of slider turning off the light. Getting to 1% brightness is going to take skill now

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-light-card.ts]: I suggest to make this change now, so we have the handle back when the light is off, and check if we can get the old behaviour of round slider back.

**Comment by homeassistanthomegear** [comment]
Are there any plans to bring back the look of pre 102 which renders all light cards the same size, even if they have dimming capabilities? Now it look quite ugly if one mixes Dimmable and non-dimmable lights in a horizontal stack.

> > I'm not sure how it behaved before, but does the height of the card change between lights that support vs. don't support brightness?
> > I.e. will it look weird if you have two of them next to each other in a horizontal stack?
> 
> Mmm yeah, it now looks like an entity button card when it doesn't support brightness...
> 
> image

Are there any plans to bring back the look of pre 102 which renders all light cards the same size, even if they have dimming capabilities? Now it look quite ugly if one mixes Dimmable and non-dimmable lights in a horizontal stack.

**Comment by thomasloven** [comment]
Please open an Issue and refer to this PR instead.

**Pull request merged**